Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1094 @@
+        $AutoRemediation,
+
+        [String]
+        [Parameter(ParameterSetName = "WetRun", HelpMessage="Specifies the time of creation of file to be used for logging remediation details when AutoRemediation switch is used")]
+        $TimeStamp
+    )
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 1 of 4] Prepare to set required TLS version for all custom domain of Front Door in Subscription: [$($SubscriptionId)]"
+    Write-Host $([Constants]::SingleDashLine)
+
+    if ($PerformPreReqCheck)
+    {
+        try
+        {
+            Write-Host "Setting up prerequisites..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            Setup-Prerequisites
+            Write-Host "Completed setting up prerequisites." -ForegroundColor $([Constants]::MessageType.Update)
+            Write-Host $([Constants]::SingleDashLine)
+        }
+        catch
+        {
+            Write-Host "Error occurred while setting up prerequisites. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+            return
+        }
+    }
+
+    # Connect to Azure account
+    $context = Get-AzContext
+
+    if ([String]::IsNullOrWhiteSpace($context))
+    {
+        Write-Host "No active Azure login session found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+        return
+    }
+
+    # Setting up context for the current Subscription.
+    $context = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop
+    
+    if(-not($AutoRemediation))
+    {
+        Write-Host "Subscription Name: [$($context.Subscription.Name)]"
+        Write-Host "Subscription ID: [$($context.Subscription.SubscriptionId)]"
+        Write-Host "Account Name: [$($context.Account.Id)]"
+        Write-Host "Account Type: [$($context.Account.Type)]"
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    Write-Host "To Set Minimum TLS version for front door in a Subscription, Contributor or higher privileges on the Front Doors are required." -ForegroundColor $([Constants]::MessageType.Warning)
+    Write-Host $([Constants]::SingleDashLine)
+    Write-Host "[Step 2 of 4] Fetch all Front Doors"
+    Write-Host $([Constants]::SingleDashLine)
+
+    $FrontDoorResourceType = "Microsoft.Network/frontDoors"
+    $FrontDoorResources = @()
+
+    # To keep track of remediated and skipped resources
+    $logRemediatedResources = @()
+    $logSkippedResources=@()
+
+    $controlIds = "Azure_FrontDoor_CDNProfile_DP_Use_Secure_TLS_Version_Trial"
+    
+    # No file path provided as input to the script. Fetch all Front Doors in the Subscription.
+    if($AutoRemediation)
+    {
+        if(-not (Test-Path -Path $Path))
+        {
+        Write-Host "File containing failing controls details [$($Path)] not found. Skipping remediation..." -ForegroundColor $([Constants]::MessageType.Error)
```

Comment:
Plz check if code should be moved by 1 tab.

Reply:
Fixed